src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -89,6 +96,18 @@ def register_options(cls, register):
                 "treated as a potential dependency if this option is set to 2 but not if set to 3."
             ),
         )
+        register(
+            "--string-resources-and-files",
```

Comment:
@Eric-Arellano what do we think about this being `resources-and-files-from-strings`. Otherwise it sounds like there'd be non-string resource and files parsed :thinking:

Reply:
Yeah I like that a lot

Reply:
I'm still not sitting on this well. Opened a discussion in Slack: https://pantsbuild.slack.com/archives/C0D7TNJHL/p1643421103775879